diff --git a/src/frontend/impl/memory_trace/readwrite_trace.cpp b/src/frontend/impl/memory_trace/readwrite_trace.cpp
index cee6a27..c9bee39 100644
--- a/src/frontend/impl/memory_trace/readwrite_trace.cpp
+++ b/src/frontend/impl/memory_trace/readwrite_trace.cpp
@@ -21,6 +21,7 @@ class ReadWriteTrace : public IFrontEnd, public Implementation {
 
     size_t m_trace_length = 0;
     size_t m_curr_trace_idx = 0;
+    bool m_finished = false;
 
     Logger_t m_logger;
 
@@ -38,8 +39,11 @@ class ReadWriteTrace : public IFrontEnd, public Implementation {
 
     void tick() override {
       const Trace& t = m_trace[m_curr_trace_idx];
-      m_memory_system->send({t.addr_vec, t.is_write ? Request::Type::Write : Request::Type::Read});
+      m_memory_system->send({t.addr_vec[0], t.is_write ? Request::Type::Write : Request::Type::Read});
       m_curr_trace_idx = (m_curr_trace_idx + 1) % m_trace_length;
+        if (m_curr_trace_idx == 0) {
+          m_finished = true;
+        }
     };
 
 
@@ -92,7 +96,7 @@ class ReadWriteTrace : public IFrontEnd, public Implementation {
 
     // TODO: FIXME
     bool is_finished() override {
-      return true; 
+      return m_finished; 
     };    
 };
 
